ament_index_python/test/test_ament_index_python.py

```diff
@@ -181,4 +196,110 @@ def get_package_share_directory_test(package_name, expect_prefix):
+        'resource_type3,  resource_type4, ' + \
+        'resource_type5 and packages'
+
+    set_ament_prefix_path(['prefix1'])
+    resources = get_resource_types()
+    assert resources == {'resource_type1',
+                         'resource_type2',
+                         'resource_type4',
+                         'resource_type5',
+                         'packages'}, 'Expected resources to ' + \
+        'be found: resource_type1,  resource_type2, ' + \
+        'resource_type4, resource_type5 and packages'
+
+
+def test_main_tool(capsys):
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+    base_path = os.path.dirname(__file__)
+
+    # Get output from the tool with name empty
+    main()
+    captured = capsys.readouterr()
+    expected_result = 'packages\n' + \
+                      'resource_type1\n' + \
+                      'resource_type2\n' + \
+                      'resource_type3\n' + \
+                      'resource_type4\n' + \
+                      'resource_type5\n'
+    assert captured.out == expected_result
+
+    # Get output from the tool with name empty
+    main(argv=['packages'])
+    captured = capsys.readouterr()
+    expected_result = 'bar\t' + base_path + os.sep + 'prefix1\n' + \
+                      'baz\t' + base_path + os.sep + 'prefix2\n' + \
+                      'foo\t' + base_path + os.sep + 'prefix1\n'
+    assert captured.out == expected_result
+
+    # Get output using both of the variables
+    main(argv=['packages', 'bar'])
+    captured = capsys.readouterr()
+    expected_result = base_path + os.sep + 'prefix1\n'
+    assert captured.out == expected_result
+
+    # Get output using both of the variables, content
+    main(argv=['resource_type4', 'foo'])
+    captured = capsys.readouterr()
+    expected_result = base_path + os.sep + 'prefix1\n'
+    expected_result = expected_result + '<<<\nfoo\n>>>\n'
+    assert captured.out == expected_result
+
+    # Try getting something not available
+    resource_type = 'packages'
+    resource_name = 'not_available'
+    result = main(argv=[resource_type, resource_name])
+    captured = capsys.readouterr()
+    expected_result = "Could not find the resource '%s' of type '%s'" \
+                      % (resource_name, resource_type)
+    assert result == expected_result
+
+
+def test_autocomplete():
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+
+    # Test with partial resource type
+    result = sorted(resource_type_completer('res'))
+    expected_result = ['resource_type1', 'resource_type2',
+                       'resource_type3', 'resource_type4',
+                       'resource_type5']
+    assert result == expected_result
+
+    # Test with partial resource name
+    class arguments():
+        resource_type = 'packages'
+    # Second argument typed (ba) is not used here
+    result = sorted(resource_name_completer('ba', arguments))
+    expected_dict = {'foo': '', 'bar': '', 'baz': ''}
```

Comment:
Should 'foo' be here?

Reply:
That's the expected result for the resource_name_completer function as it is now: it ignores the 'ba' argument. Not sure if this was done on purpose (because using it may result in two searches and it might take a while to return anything in the cli) or it is a functionality bug. Let me know your thoughts.

Reply:
Yeah, it's a bug. Can you send a separated PR fixing that? Then you can rebase and update the test here.
It should add some filtering to `resource_name_completer`, similar to what's done here:
https://github.com/ament/ament_index/blob/9d42ba13d7694ad8da5b1622e433e691996c4502/ament_index_python/ament_index_python/cli.py#L67